File: public/assets/pythonCode/insertionsort.py
import time
import random

def insertionSort(numbers: list[int]) -> None:
    n: int = len(numbers)
    for i in range(1, n):
        element: int = numbers[i]
        j: int = i - 1
        while j >= 0 and element < numbers[j]:
            numbers[j + 1] = numbers[j]
            j -= 1
        numbers[j + 1] = element


amount = 10000
# You can change this to see how long it take to completely sort the array
arr = [random.randint(1, 1000) for i in range(amount)]
start = time.time()
insertionSort(arr)
end = time.time()

print(arr)
print(f"Time taken: {end - start:.6f} seconds")

File: public/assets/javaCode/insertionsort.js
function insertionSort(numbers) {
    const n = numbers.length;

    for (let i = 1; i < n; i++) {
        const element = numbers[i];
        let j = i - 1;

        while (j >= 0 && element < numbers[j]) {
            numbers[j + 1] = numbers[j];
            j--;
        }
        numbers[j + 1] = element;
    }
}

// Number of elements
const amount = 10000;


const arr = Array.from({ length: amount }, () =>
    Math.floor(Math.random() * 1000) + 1
);

const start = performance.now();

insertionSort(arr);
const end = performance.now();

console.log(arr);
console.log(`Time taken: ${(end - start).toFixed(6)} ms`);